### PR TITLE
V22F-904 Add threshold button to researcher menu

### PR DESCRIPTION
**Pull request by shoekzema**
Also fix the default threshold button bug in the dev menu AND improve the keyboard-button-controls

### CODE REVIEW

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/DeveloperMenu.cpp [hunk 1]
```diff
@@ -45,8 +42,8 @@ int m_accelThresholdControl;
 int m_brakeThresholdControl;
 int m_steerThresholdControl;
 
-// Control for the default values
-int m_defaultButton;
+// The current interventionType set in the researcher menu
+int m_tempInterventionType;
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/DeveloperMenu.cpp:46]:
```suggestion
InterventionType m_tempInterventionType;
```

File: speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/DeveloperMenu.cpp [hunk 2]
```diff
@@ -308,6 +305,20 @@ static void SetDefaultThresholdValues(void*)
...
+/// @brief Set the default values of threshold boxes based on the InterventionType.
+void RemoteSetDefaultThresholdValues()
+{
+    SetDefaultThresholdValues(nullptr);
+    SaveSettings();
+}
+
+/// @brief                    Set the temporary interventionType to p_interventionType
+/// @param p_interventionType The intervention type
+void SetTempInterventionType(int p_interventionType)
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/DeveloperMenu.cpp:317]:
```suggestion
void SetTempInterventionType(InterventionType p_interventionType)
```

**Review by suppergerrie2 — CHANGES_REQUESTED** [review]
Add changes to the code changes document

**Review by larsdekwant — APPROVED** [review]
user feedback might be a feature for later, the functionality works.

**Pull request merged**